spark/src/main/scala/io/ddf/spark/content/ViewHandler.scala

```diff
@@ -51,17 +59,41 @@ class ViewHandler(mDDF: DDF) extends io.ddf.content.ViewHandler(mDDF) with IHand
     this.get(columns, ViewFormat.withName(format))
   }
 
-  override  def getRandomSampleByNum(numSamples: Int, withReplacement: Boolean,
-    seed:
-  Int): DDF = {
-    null.asInstanceOf[DDF]
+  override def getRandomSampleByNum(numSamples: Long, withReplacement: Boolean,
+                                    seed:
+                                    Int): DDF = {
+
+    if (numSamples < 0) {
+      throw new IllegalArgumentException("Number of samples must be larger than or equal to 0")
```

Comment:
without replacement, it should be [1, ddf.nrow], with replacement it should be [1, 2 * ddf.nrow] as the limitation of current implementation